### DISCUSSION TITLE
Programmatically Triggering Click Event

### DISCUSSION THREAD

**Discussion by o12486vs2** [zauberzeug/nicegui#1321] [Q&A]
### Question

I'm trying to programmatically trigger a click event on an element without actually clicking it. I have attempted to use run_javascript to apply the following methods:

```js
document.getElementById("myInput").focus();
document.getElementById("myInput").click();
document.getElementById("myInput").blur();
```
Unfortunately, none of these methods seem to produce the desired effect. Could you suggest any potential solutions or alternatives that could help me achieve this?

**Comment by falkoschindler** [comment]
Here is an example of a button `b` with an event handler `handle_click`. A second button calls the JavaScript method "click" to trigger a click event programmatically.

```py
def handle_click():
    ui.notify('You clicked the button!')

b = ui.button('Click me', on_click=handle_click)

ui.button('Trigger click', on_click=lambda: b.run_method('click'))
```

But I wonder why you don't simply call the event handler `handle_click` programmatically instead of making the detour over the client?
